src/application/Application.mjs

```diff
@@ -405,6 +421,195 @@ export default class Application extends Component {
+                    this.fireBottomUpScrollWheelHandler("_handleScrollDown", obj);
+                }
+            } else if (deltaY < 0) {
+                if (!this.fireTopDownScrollWheelHandler("_captureScrollUp")) {
+                    this.fireBottomUpScrollWheelHandler("_handleScrollUp", obj);
+                }
+            }
+        }
+    }
+
+    fireTopDownScrollWheelHandler(event) {
+        let children = this.stage.application.children;
+        let affected = this._findChildren([], children).reverse();
+        let n = affected.length;
+
+        while(n--) {
+            const child = affected[n];
+            if (child && child[event]) {
+                if (event === "_captureScrollDown") {
+                    child._captureScrollDown();
+                    return true;
+                } else if (event === "_captureScrollUp") {
+                    child._captureScrollUp();
+                    return true;;
+                }         
+            }
+        }
+        return false
+    }
+
+    fireBottomUpScrollWheelHandler(event, obj) {
+        const {clientX, clientY} = obj;
+        const target = this._getTargetChild(clientX, clientY);
+        let child = target;
+
+        // Search tree bottom up for a handler
+        while (child !== null) {
+            if (child && child[event]) {
+                if (event === "_handleScrollDown") {
+                    child._handleScrollDown();
+                    return true;
+                } else if (event === "_handleScrollUp") {
+                    child._handleScrollUp();
+                    return true;;
+                }         
+            }
+            child = child.parent;
+        }
+        return false
+    }
+
+    _receiveClick(e) {
+        const obj = e;
+        const {clientX, clientY} = obj;
+
+        if (clientX <= this.stage.w && clientY <= this.stage.h) {
+            // force a drawFrame to set any missing element dimensions
+            this.stage.drawFrame();
+            this.stage.application.fireBottomUpClickHandler(obj);
+        }
+    }
+
+    fireBottomUpClickHandler(obj) {
+        const {clientX, clientY} = obj;
+        const target = this._getTargetChild(clientX, clientY);
+        let child = target;
+
+        // Search tree bottom up for a handler
+        while (child !== null) {
+            if (child && child["_handleClick"]) {
+                child._handleClick(target);
+                break;
+            }
+            child = child.parent;
+        }
+    }
+
+    _receiveHover(e) {
+        const obj = e;
+        const {clientX, clientY} = obj;
+
+        if (clientX <= this.stage.w && clientY <= this.stage.h) {
+            // force a drawFrame to set any missing element dimensions
+            this.stage.drawFrame();
+            this.stage.application.fireBottomUpHoverHandler(obj);
+        }
+    }
+
+    fireBottomUpHoverHandler(obj) {
```

Comment:
It would be useful to provide `_handleUnhover` hook as well, e.g:

```javascript
    fireBottomUpHoverHandler(obj) {
        const {clientX, clientY} = obj;
        const target = this._getTargetChild(clientX, clientY);

        // Only fire handler when pointer target changes
        if (target !== this.__hoveredChild) {
            if (this.__hoveredChild) {
                let child = this.__hoveredChild;
                while (child !== null) {
                    if (child && child["_handleUnhover"]) {
                        child._handleUnhover(target);
                        break;
                    }
                    child = child.parent;
                }
            }

            let child = target;
            this.__hoveredChild = target;

            while (child !== null) {
                if (child && child["_handleHover"]) {
                    child._handleHover(target);
                    break;
                }
                child = child.parent;
            }
        }
    }
```